src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -194,7 +194,7 @@ private unsafe Socket(SafeSocketHandle handle, bool loadPropertiesFromHandle)
                         {
                             // GetPeerName() bellow will fail if given buffer is too small.
                             // Give it at least old Windows max path.
```

Comment:
This comment is no longer correct.

Reply:
what text do you suggest? Should I put something down bellow to retry/realloc block as well?

Reply:
Yes, I think that's a good idea.  Basically here don't special-case Windows, and then if we get the relevant failure later, allocate the max buffer needed to try again.  If that then fails, bail.  Or something along those lines.